### PR TITLE
Update service connection and other parameters for ESRP v5

### PR DESCRIPTION
**Pull request by Sohamdg081992**
[comment]: # (Note that your PR title should follow the conventional commit format: https://conventionalcommits.org/en/v1.0.0/#summary)
# PR Description
1. Update service connection and other parameters for ESRP v5
2. Update to sign all images

Doc followed for the new ESRP setup: [MS Sign Knowledge Base - ADO Task v5 (sharepoint.com)](https://microsoft.sharepoint.com/teams/prss/Codesign/SitePages/ADO%20Task%20v5.aspx?OR=Teams-HL&CT=1714049743060&xsdata=MDV8MDJ8c29oZGFzZ3VwdGFAbWljcm9zb2Z0LmNvbXw4YTc4ZGVmNDJkODg0MjJmNTM4ZDA4ZGM2NTQwMDU4NHw3MmY5ODhiZjg2ZjE0MWFmOTFhYjJkN2NkMDExZGI0N3wxfDB8NjM4NDk2NTcyODU0MjA1ODAwfFVua25vd258VFdGcGJHWnNiM2Q4ZXlKV0lqb2lNQzR3TGpBd01EQWlMQ0pRSWpvaVYybHVNeklpTENKQlRpSTZJazFoYVd3aUxDSlhWQ0k2TW4wPXwwfHx8&sdata=aHV6OG95VHVCbmc4OE9ETkN2bGRWMndMZHh0Y0R3WDdIUkRVWnF3ZG5jTT0%3d&clickparams=eyAiWC1BcHBOYW1lIiA6ICJNaWNyb3NvZnQgT3V0bG9vayIsICJYLUFwcFZlcnNpb24iIDogIjE2LjAuMTc1MzEuMjAwOTAiLCAiT1MiIDogIldpbmRvd3MiIH0%3D)
[comment]: # (The below checklist is for PRs adding new features. If a box is not checked, add a reason why it's not needed.)
# New Feature Checklist

- [ ] List telemetry added about the feature.
- [ ] Link to the one-pager about the feature.
- [ ] List any tasks necessary for release (3P docs, AKS RP chart changes, etc.) after merging the PR.
- [ ] Attach results of scale and perf testing.

[comment]: # (The below checklist is for code changes. Not all boxes necessarily need to be checked. Build, doc, and template changes do not need to fill out the checklist.)
# Tests Checklist

- [ ] Have end-to-end Ginkgo tests been run on your cluster and passed? To bootstrap your cluster to run the tests, follow [these instructions](/otelcollector/test/README.md#bootstrap-a-dev-cluster-to-run-ginkgo-tests).
  - Labels used when running the tests on your cluster:
    - [ ] `operator`
    - [ ] `windows`
    - [ ] `arm64`
    - [ ] `arc-extension`
- [ ] Have new tests been added? For features, have tests been added for this feature? For fixes, is there a test that could have caught this issue and could validate that the fix works?
  - [ ] Is a new scrape job needed?
    - [ ] The scrape job was added to the folder [test-cluster-yamls](/otelcollector/test/test-cluster-yamls/) in the correct configmap or as a CR. 
  - [ ] Was a new test label added?
    - [ ] A string constant for the label was added to [constants.go](/otelcollector/test/utils/constants.go).
    - [ ] The label and description was added to the [test README](/otelcollector/test/README.md).
    - [ ] The label was added to this [PR checklist](/.github/pull_request_template).
    - [ ] The label was added as needed to [testkube-test-crs.yaml](/otelcollector/test/testkube/testkube-test-crs.yaml).
  - [ ] Are additional API server permissions needed for the new tests?
    - [ ] These permissions have been added to [api-server-permissions.yaml](/otelcollector/test/testkube/api-server-permissions.yaml).
  - [ ] Was a new test suite (a new folder under `/tests`) added?
    - [ ] The new test suite is included in [testkube-test-crs.yaml](/otelcollector/test/testkube/testkube-test-crs.yaml).

### CODE REVIEW

File: .pipelines/azure-pipeline-build.yml [hunk 1]
```diff
@@ -416,10 +417,14 @@ stages:
         displayName: "Build: Set values in payload.json for signing"
         condition: and(succeeded(), eq(variables.IS_MAIN_BRANCH, true))
 
-      - task: EsrpCodeSigning@3
-        displayName: "ESRP CodeSigning for Prometheus"
+      - task: EsrpCodeSigning@5
```
Comment (bragi92) [inline .pipelines/azure-pipeline-build.yml:420]: revert the change branch name change (esrpUpdateSvcConnection) on line 5 and keep the display name of the tasks?

Do we have to sign the reference app image too or is the guidance just for images we serve to customers?

**Review by vishiy — DISMISSED** [review]

File: .pipelines/azure-pipeline-build.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ trigger:
   branches:
     include:
     - main
+    - esrpUpdateSvcConnection
```
Comment (bragi92) [inline .pipelines/azure-pipeline-build.yml:5]: revert the branch name?

Comment (bragi92) [inline .pipelines/azure-pipeline-build.yml:5]: revert the branch name? looks like the ESRP build step is failing?

**Review by bragi92 — APPROVED** [review]
